src/components/Common/NavBar/__tests__/Navbar.test.js

```diff
@@ -0,0 +1,120 @@
+import App from "App";
+import RestService from "utils/services/rest";
+import { usersData, itemsData, categoriesData } from "tests/fixtures/database";
+import { loadState } from "utils/services/localStorage";
+import { createMockedState } from "tests/fixtures/state";
+
+jest.mock("utils/services/rest", () => ({
+  getWithToken: jest.fn(),
+  get: jest.fn(),
+}));
+
+const mockedUser1State = createMockedState(1);
+
+beforeEach(() => {
+  // Mock APIs
+  // - Category & item APIs
+  RestService.get.mockImplementation(async (url, configs) => {
+    const urlObj = new URL(url);
+    const paramsList = urlObj.pathname.split("/");
+    // Item detail
+    if (urlObj.pathname.match(/\/categories\/[\d]*\/items\/[\d]*/)) {
+      const [, , categoryId, , itemId] = paramsList;
+      // Get items list of specific category
+      const itemDetail = itemsData.items
+        .filter(
+          (item) =>
+            item.categoryId === parseInt(categoryId) &&
+            item.id === parseInt(itemId)
+        )
+        .map((item) => ({
+          ...item,
+          author: {
+            id: item.authorId,
+            name: usersData.info[item.authorId].name,
+          },
+        }))[0];
+      return Promise.resolve({ ...itemDetail });
+    }
+    // Items list
+    else if (urlObj.pathname.match(/\/categories\/[\d]*\/items/)) {
+      const [offset, limit] = urlObj.searchParams.values();
+      const [, , categoryId] = paramsList;
+
+      // Get items list of specific category
+      const itemsList = itemsData.items
+        .filter((item) => item.categoryId === parseInt(categoryId))
+        .map((item) => ({
+          ...item,
+          author: {
+            id: item.authorId,
+            name: usersData.info[item.authorId].name,
+          },
+        }));
+      return Promise.resolve({
+        items: itemsList.slice(offset, offset + limit),
+        totalItems: itemsList.length,
+      });
+    }
+    // Categories detail
+    else if (urlObj.pathname.match(/\/categories\/[\d]*/)) {
+      const [, , categoryId] = paramsList;
+
+      const categoryDetail = categoriesData.items.filter(
+        (category) => category.id === parseInt(categoryId)
+      )[0];
+      return Promise.resolve({ ...categoryDetail });
+    }
+    // Category list
+    else {
+      const [offset, limit] = urlObj.searchParams.values();
+      const categoriesList = categoriesData.items;
+      return Promise.resolve({
+        items: categoriesList.slice(offset, offset + limit),
+        totalItems: categoriesList.length,
+      });
+    }
+  });
+
+  // - Get user information API
+  RestService.getWithToken.mockImplementation(async (url, configs) => {
+    const token = loadState()?.user.token;
+    const userId = usersData.token[token];
+    return userId
+      ? Promise.resolve({ name: usersData.info[userId].name, id: userId })
+      : Promise.reject({});
+  });
+});
+
+afterEach(() => {
+  global.localStorage.clear();
+});
+
+describe("navigation", () => {
```

Comment:
This is an unit test. The name covention is the path to the implementstion file. Ex: "components/common/Navbar"